tests/integration/sandbox.py

```diff
@@ -59,3 +60,24 @@ def test_remote_apis_socket(cli, datafiles):
 
     result = cli.run(project=project, args=["build", element_name])
     assert result.exit_code == 0
+
+
+# Test configuration with remote action cache for nested REAPI.
+@pytest.mark.skipif(not HAVE_SANDBOX, reason="Only available with a functioning sandbox")
+@pytest.mark.datafiles(DATA_DIR)
+def test_remote_apis_socket_with_action_cache(cli, tmpdir, datafiles):
+    project = str(datafiles)
+    element_name = "sandbox/remote-apis-socket.bst"
+
+    with create_artifact_share(os.path.join(str(tmpdir), "remote")) as share:
+        cli.configure(
+            {
+                "remote-execution": {
+                    "storage-service": {"url": share.repo},
+                    "action-cache-service": {"url": share.repo, "push": True},
```

Comment:
Could you please add tests for the other possible configurations? (cache storage-service with remote-execution action-cache-service, and both storage services with remote-execution action-cache-service)

Slightly related, considering the case where both the `cache` and `remote-execution` storage services are configured, does this lead to a sensible outcome? (e.g. are blobs downloaded from the remote-execution storage uploaded to the cache storage?)

Reply:
I've added these configuration tests.

Upload to cache storage-service was missing in case two different CAS remotes are configured. I've added this, reusing code we already had for remote execution.